mlir/lib/Dialect/Vector/Transforms/VectorDropLeadUnitDim.cpp

```diff
@@ -399,13 +399,29 @@ mlir::vector::castAwayContractionLeadingOneDim(vector::ContractionOp contractOp,
           transposeResults.push_back(targetExpr);
         }
       }
+
+      // Checks if only the outer, unit dimensions (of size 1) are permuted.
+      // Such transposes do not materially effect the underlying vector and can
+      // be omitted. EG: perm [1, 0, 2] applied to vector<1x1x8xi32>
+      bool tranposeNonOuterUnitDims = false;
+      for (auto [index, dim] :
```

Comment:
Thanks for the example, I now understand why we need this.. Instead of adding the ad-hoc logic here, would it make sense to add the canonicalization pattern to vector.transpose op?

Reply:
SGTM

Reply:
The follow up SGTM as well. Having that canonicalization pattern makes sense but given that we already have logic to decide if a transpose should be generated or not, it would make sense to extend that logic to support this extra case (and hopefully avoid a canonicalization pass just for this).

Reply:
going to commit this for now and start work on a broader transpose fold